### PR TITLE
Show results from pre-tested commits from the same base branch in the performance report

### PR DESCRIPTION
**Pull request by wolfs**
Since we currently only run performance tests on pre-tested commit branches and not on `master` any more, the performance test reports are a little bit boring, since they don't show much history.

We could instead show the history of all the pre-tested commits build for the same base branch.

This PR tries this by mapping the pre-tested commit branches to the base branch. It would be nicer though to show the actual branch instead of the base branch in the list of changes.

### CODE REVIEW

**Review by blindpirate — COMMENTED** [review]
I left two tiny comments.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/CrossVersionResultsStore.java [hunk 1]
```diff
@@ -225,103 +226,122 @@ private void addOperations(PreparedStatement statement, long testId, String vers
...
-                            } else {
-                                BaselineVersion baselineVersion = result.baseline(version);
-                                baselineVersion.getResults().add(operation);
-                                allVersions.add(version);
-                            }
+                        String version = operations.getString(1);
+                        if ("1.7".equals(version) && result.getStartTime() <= ignoreV17Before) {
+                            // Ignore some broken samples
+                            continue;
+                        }
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/CrossVersionResultsStore.java]: I think so. Removed.

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Comment by wolfs** [comment]
@bot-gradle cancel

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**